File: Episode 3 Trees I 树 （上）/3.3 Binary Tree Traversal 遍历二叉树/3.3 Binary Tree Traversal.cpp
//Note that for all of the 3 traversal types, the path of traversal are the same, the differences lie in when do you print their values:

void PreOrderTraversal(BinTree BT)
{
	if (BT){
		printf("%d",BT->Data);
		PreOrderTraversal(BT->Left);
		PreOrderTraversal(BT->Right);
	}
}

void InOrderTraversal(BinTree BT)
{
	if (BT){
		InOrderTraversal(BT->Left);
		printf("%d",BT->Data);
		InOrderTraversal(BT->Right);
	}
}

void PostOrderTraversal(BinTree BT)
{
	if (BT){
		PostOrderTraversal(BT->Left);
		PostOrderTraversal(BT->Right);
		printf("%d",BT->Data);
	}
}

File: Episode 3, 4 Trees I 树 （上）（下）/3.3 Binary Tree Traversal 遍历二叉树/3.3 Binary Tree Traversal.py
# Python Version

def PreOrderTraversal(root):

	print(root.val)
	if root.left:
		PreOrderTraversal(root.left)
	if root.right
		PreOrderTraversal(root.right)
	return

def InOrderTraversal(root):

	if root.left:
		InOrderTraversal(root.left)
	print(root.val)
	if root.right
		InOrderTraversal(root.right)
	return

def PostOrderTraversal(root):

	if root.left:
		PostOrderTraversal(root.left)
	if root.right:
		PostOrderTraversal(root.right)
	print(root.val)
	return

